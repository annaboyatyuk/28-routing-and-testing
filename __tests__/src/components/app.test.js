import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../../src/components/app/index';
import NoteCreate from '../../../src/components/note-create-form/index';
import Dashboard from '../../../src/components/dashboard/index';



xdescribe('<App/> (Enzyme Test)', () => {
  it('is alive at application start', () => {
    let app = shallow(<App/>);
    expect(app.find('main').exists()).toBeTruthy();
  });
});


describe('<App/> (Snapshot Test', () => {
  it('renders', () => {
    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Note create form snapshot test', () => {
    const component = renderer.create(
      <NoteCreate />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('Dashboard form snapshot test', () => {
    const component = renderer.create(
      <Dashboard />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

