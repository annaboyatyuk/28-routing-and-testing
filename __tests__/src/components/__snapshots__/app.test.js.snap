// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> (Snapshot Test Dashboard form snapshot test 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label>
      TITLE
    </label>
    <input
      id="title"
      onChange={[Function]}
      type="text"
    />
    <label>
      NOTE
    </label>
    <input
      id="content"
      onChange={[Function]}
      type="text"
    />
    <input
      type="submit"
    />
  </form>,
  <ul />,
]
`;

exports[`<App/> (Snapshot Test Note create form snapshot test 1`] = `
<form>
  <label>
    TITLE
  </label>
  <input
    id="title"
    type="text"
  />
  <label>
    NOTE
  </label>
  <input
    id="content"
    type="text"
  />
  <input
    type="submit"
  />
</form>
`;

exports[`<App/> (Snapshot Test renders 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          HOME
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/dashboard"
          onClick={[Function]}
        >
          DASHBOARD
        </a>
      </li>
    </ul>
  </nav>,
  <main>
    <h2>
      Welcome to the Notes
    </h2>
    <p>
      Add notes to this app! delete them if you get sick of them! its fun!
    </p>
    <p>
      click the dashboard to get startedddd
    </p>
  </main>,
]
`;
